controllers/processingjob/processingjob_controller.go

```diff
@@ -0,0 +1,306 @@
+limitations under the License.
+*/
+
+package processingjob
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	processingjobv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/processingjob"
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers"
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers/sdkutil"
+	"github.com/aws/amazon-sagemaker-operator-for-k8s/controllers/sdkutil/clientwrapper"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sagemaker"
+)
+
+const (
+	// ReconcilingProcessingJobStatus is the status used by the controller during reconciliation.
+	ReconcilingProcessingJobStatus = "ReconcilingProcessingJob"
+
+	// MaxProcessingJobNameLength defines the maximum number of characters in a SageMaker Processing Job name
+	MaxProcessingJobNameLength = 63
+)
+
+// Reconciler reconciles a ProcessingJob object
+type Reconciler struct {
+	client.Client
+	Log                   logr.Logger
+	PollInterval          time.Duration
+	createSageMakerClient clientwrapper.SageMakerClientWrapperProvider
+	awsConfigLoader       controllers.AwsConfigLoader
+}
+
+// NewProcessingJobReconciler creates a new reconciler with the default SageMaker client.
+func NewProcessingJobReconciler(client client.Client, log logr.Logger, pollInterval time.Duration) *Reconciler {
+	return &Reconciler{
+		Client:       client,
+		Log:          log,
+		PollInterval: pollInterval,
+		createSageMakerClient: func(cfg aws.Config) clientwrapper.SageMakerClientWrapper {
+			return clientwrapper.NewSageMakerClientWrapper(sagemaker.New(cfg))
+		},
+		awsConfigLoader: controllers.NewAwsConfigLoader(),
+	}
+}
+
+// +kubebuilder:rbac:groups=sagemaker.aws.amazon.com,resources=processingjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=sagemaker.aws.amazon.com,resources=processingjobs/status,verbs=get;update;patch
+
+// Reconcile attempts to reconcile the SageMaker resource state with the k8s desired state.
+func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := reconcileRequestContext{
+		Context:       context.Background(),
+		Log:           r.Log.WithValues("processingjob", req.NamespacedName),
+		ProcessingJob: new(processingjobv1.ProcessingJob),
+	}
+
+	ctx.Log.Info("Getting resource")
+	if err := r.Get(ctx, req.NamespacedName, ctx.ProcessingJob); err != nil {
+		ctx.Log.Info("Unable to fetch ProcessingJob job", "reason", err)
+
+		if apierrs.IsNotFound(err) {
+			return controllers.NoRequeue()
+		}
+
+		return controllers.RequeueImmediately()
+	}
+
+	if err := r.reconcileProcessingJob(ctx); err != nil {
+		ctx.Log.Info("Got error while reconciling, will retry", "err", err)
+		return controllers.RequeueImmediately()
```

Comment:
do you know if this backs off re-queuing with time ? We were supposed to verify but I haven't

Reply:
I believe all requeue events are automatically backed off.